### PR TITLE
Install and infrastructure fixes

### LINKED ISSUE #1648: Update machine installation for Toolbox
Machine is now installed via Toolbox on Mac/Windows

### PR DESCRIPTION
**Pull request by moxiegirl**
- Updating install docs fixes #1648
- Updating the Dockerfile to lastest
- Adding support for building all the public doc projects from machine

Signed-off-by: Mary Anthony <mary@docker.com>

ping @jeffdm @mchiang0610

### CODE REVIEW

File: docs/install-machine.md [hunk 1]
```diff
@@ -17,61 +17,33 @@ architectures are available at the [Github
...
 
-```
-$ docker-machine -v
-machine version 0.3.1
-```
+2. Download the Machine binary to somewhere in your `PATH` (for example, `/usr/local/bin`).
 
-In order to run Docker commands on your machines without having to use SSH, make
-sure to install the Docker client as well, e.g.:
+        $ curl -L https://github.com/docker/machine/releases/download/v0.3.1/docker-machine_linux-amd64 > /usr/local/bin/docker-machine
```
Comment (mchiang0610) [inline docs/install-machine.md]: Do we update them to v.0.4.0 that's coming out next week?

File: docs/Dockerfile [hunk 2]
```diff
@@ -1,26 +1,27 @@
 FROM docs/base:latest
 MAINTAINER Mary Anthony <mary@docker.com> (@moxiegirl)
 
-# to get the git info for this repo
+# To get the git info for this repo
 COPY . /src
 
 COPY . /docs/content/machine/
 
+RUN svn checkout https://github.com/docker/compose/trunk/docs /docs/content/compose
```
Comment (mchiang0610) [inline docs/Dockerfile]: Just curious on why we're using SVN

File: docs/get-started.md [hunk 3]
```diff
@@ -110,7 +110,7 @@ $ docker-machine ip dev
 192.168.99.100
 ```
 
-For instance, you can try running a webserver ([nginx](https://nginx.org)) in a
+For instance, you can try running a webserver ([nginx](https://www.nginx.com/) in a
```
Comment (mchiang0610) [inline docs/get-started.md]: awesome catch! Thanks

File: docs/install-machine.md [hunk 4]
```diff
@@ -17,61 +17,33 @@ architectures are available at the [Github
...
 
 
-### OS X and Linux
+## OS X and Windows
 
-To install on OS X or Linux, download the proper binary to somewhere in your
-`PATH` (e.g. `/usr/local/bin`) and make it executable. For instance, to install on
-most OS X machines these commands should suffice:
+Install Machine using the Docker Toolbox using the <a href="https://docs.docker.com/installation/mac/" target="_blank">Mac OS X installation</a>
+instruction or <a href="https://docs.docker.com/installation/windows" target="_blank">windows installation</a> instructions.
```
Comment (mchiang0610) [inline docs/install-machine.md]: Can we capitalize the W -> "Windows installation"

File: docs/install-machine.md [hunk 5]
```diff
@@ -17,61 +17,33 @@ architectures are available at the [Github
...
-$ curl -L https://github.com/docker/machine/releases/download/v0.3.1/docker-machine_darwin-amd64 > /usr/local/bin/docker-machine
-$ chmod +x /usr/local/bin/docker-machine
-```
+## On Linux
 
-For Linux, just substitute "linux" for "darwin" in the binary name above.
+To install on Linux, do the following:
 
-Now you should be able to check the version with `docker-machine -v`:
+1. Install <a href="https://docs.docker.com/installation/" target="_blank">Docker Docker version 1.7.1 or greater</a>:
```
Comment (mchiang0610) [inline docs/install-machine.md]: Double Docker here.

**Comment by mchiang0610** [comment]
Otherwise, looks good to me. Adding machine maintainer @nathanleclaire to double check.

File: docs/Dockerfile (hunk 2 above)
Comment (moxiegirl) [inline docs/Dockerfile]: It allows me to checkout just a directory instead of the whole repo.

**Comment by moxiegirl** [comment]
@mchiang0610 comments in.  I'll rebase after the next reviewer.

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+The `[menu.main]` section refers to navigation defined [in the main Docker menu](https://github.com/docker/docs-base/blob/hugo/config.toml). This metadata says *add a menu item called* Extending services in Compose *to the menu with the* `smn_workdw_compose` *identifier*.  If you locate the menu in the configuration, you'll find *Create multi-container applications* is the menu title.
+
+You can move an article in the tree by specifying a new parent. You can shift the location of the item by changing its weight.  Higher numbers are heavier and shift the item to the bottom of menu. Low or no numbers shift it up.
+
+
+## Other key documentation repositories
+
+The `docker/docs-base` repository contains [the Hugo theme and menu configuration](https://github.com/docker/docs-base). If you open the `Dockerfile` you'll see the `make docs` relies on this as a base image for building the Compose documentation.
+    
+The `docker/docs.docker.com` repository contains [build system for building the Docker documentation site](https://github.com/docker/docs.docker.com). Fork this repository to build the entire documentation site.
```
Comment (thaJeztah) [inline docs/README.md]: s/contains/contains the/

**Comment by thaJeztah** [comment]
LGTM (one micro-nit to save my reputation :-))

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
+# Contributing to the Docker Compose documentation
+
+The documentation in this directory is part of the [this documentation](docs.docker.com).  Docker uses [the Hugo static generator](http://gohugo.io/overview/introduction/) to convert project Markdown files to a static HTML site.
```
Comment (nathanleclaire) [inline docs/README.md]: Are we not being strict about the 80-characters per line rule anymore?

Comment (thaJeztah) [inline docs/README.md]: I'm still +1 for 80-chars (easier reviewing and to read in plain text) 😶

Comment (nathanleclaire) [inline docs/README.md]: Yeah, same.

**Comment by moxiegirl** [comment]
@nathanleclaire @thaJeztah that  80 character info is in the contributor guide...I've ref'd the full guide and mentioned here too.

@nathanleclaire are you L G T M?

**Comment by nathanleclaire** [comment]
OK.  I was asking about `docs/README.md` here, which doesn't seem to have the 80 character-per-line wrap.  I assumed that it would apply to all of our docs.  LGTM aside from that.

**Comment by moxiegirl** [comment]
@nathanleclaire Oh, that's a case of the shoemacher's kids.   I'll pick that up in another round.

**Pull request merged**